### ISSUE TITLE
Error initializing firewall rules! 

### ISSUE THREAD

**Issue by Algogator** [nodogsplash/nodogsplash#586]
> [5][Tue Sep 08 19:58:26 2020][755](src/main.c:359) Starting as daemon, forking to background
> [5][Tue Sep 08 19:58:26 2020][756](src/main.c:273) Detected gateway wlan0 at 192.168.10.1 (00:25:ca:14:2a:9e)
> root@matma-imx6ul-var-dart:~# [5][Tue Sep 08 19:58:26 2020][756](src/main.c:289) Created web server on 192.168.10.1:2050
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> [5][Tue Sep 08 19:58:27 2020][756](src/fw_iptables.c:380) Initializing firewall rules
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> iptables: No chain/target/match by that name.
> [3][Tue Sep 08 19:58:29 2020][756](src/main.c:306) Error initializing firewall rules! Cleaning up
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> iptables v1.6.2: can't initialize iptables table `mangle': Table does not exist (do you need to insmod?)
> Perhaps iptables or your kernel needs to be upgraded.
> [3][Tue Sep 08 19:58:31 2020][756](src/main.c:308) Exiting because of error initializing firewall rules

I cloned and built the code from github

> var-dart:~#  iptables -V
> iptables v1.6.2
> var-dart:~#  uname -r
> 4.14.78+ge692c3062368

**Comment by bluewavenet** [comment]
@Algogator 
You could try to be a little more forthcoming rather than just dumping a log. You do realise there are many types of hardware, operating systems and versions of NoDogSplash?

**Comment by Algogator** [comment]
@bluewavenet 
Linux version 4.14.78+ge692c3062368 (oe-user@oe-host) (gcc version 8.2.0 (GCC)) 
Like I said I just cloned the github repo today and built it so it's 5.0.0
I'm not sure what you mean by hardware?

**Comment by bluewavenet** [comment]
@Algogator 
By hardware, I mean manufacturer/model. Is it a router? Is it an SBC? An IoT device? What architecture? How much ram? Flash?
Yes you are running kernel 4.14.78, but what distro and distro version?

**Comment by Algogator** [comment]
It's custom (yocto) on a DART-6UL, armv7l, 512 MB (http://variwiki.com/index.php?title=DART-6UL)

**Comment by bluewavenet** [comment]
@Algogator 
I've not seen a DART-6UL before. It has the limitation of a Broadcom chip, just like the Rpi and just like the Rpi it is of limited use as a router from the wireless point of view.

Your problem might be iptables version. I don't recall exactly, but a few years ago changes in iptables made it necessary to change NoDogSplash. I do not know the version number off hand.

I am on OpenWrt kernel 5.4.60, iptables 1.8.4. with iptables-mod-conntrack-extra and iptables-mod-ipopt installed 

You do need conntrack-extra and ipopt support in iptables.
These are either an option when compiling iptables or separate modules.

In summary, your problem is most likely either iptables missing support as above or iptables version (or both).

**Comment by bluewavenet** [comment]
@Algogator 
Can you upgrade to Yocto Zeus?

**Comment by bsteiner-dandy** [comment]
I am having a similar issue on a Coral Development board. Its running Mendel Linux (4.14.98-imx) and I cloned the repo today. Here is the output:
```
[5][Mon Mar 21 19:43:27 2022][5186](src/main.c:359) Starting as daemon, forking to background
[5][Mon Mar 21 19:43:27 2022][5187](src/main.c:273) Detected gateway wlan0 at 192.168.0.1 (7c:d9:5c:b1:8d:eb)
[5][Mon Mar 21 19:43:27 2022][5187](src/main.c:289) Created web server on 192.168.0.1:2050
[5][Mon Mar 21 19:43:27 2022][5187](src/fw_iptables.c:380) Initializing firewall rules
[3][Mon Mar 21 19:43:28 2022][5187](src/main.c:306) Error initializing firewall rules! Cleaning up
[3][Mon Mar 21 19:43:28 2022][5187](src/main.c:308) Exiting because of error initializing firewall rules
```

**Comment by bluewavenet** [comment]
@bsteiner-dandy 
Nodogsplash has been unmaintained for 2 years.
You might consider changing to openNDS as it is being actively developed and you will be more likely to get help there.
See the Nodogsplash README.md here:
https://github.com/nodogsplash/nodogsplash#split-of-nodogsplash
